### DISCUSSION TITLE
Outdated!  Fun features - Scenarios, object spawn & ped object attachment

### DISCUSSION THREAD

**Discussion by lonelybud** [YimMenu/YimMenu#1918] [Ideas]
# Outdated!
# Dont use any build from this discussion. 

I abandoned this discussion long ago since the ban. Code still working but i personally dont use it or recommend using. If one is looking to make lua script out if it can check [here](https://github.com/YimMenu/YimMenu/discussions/1918#discussioncomment-8378490) though. Good luck

add new features in lua script or native-

- [x] scenarios
- [x] object spawn (with ability to rotate, move, keep track of object). Example spawning ramps, table, chairs, beachfire etc. 
- ped attachment with objects. . 
  - [x] Example lifting a car with hands
  - [x] Holding glow sticks while dancing (https://www.youtube.com/watch?v=NlFejTPi-JY)
- [x] custom tab for player particles effects.
- [ ] ability to save current animation, scenario & custom player coordinates. also save spawned peds and their attachments along.
       (can we create xml instead somehow and then use them instead of JSON?)
       <details>
         <summary>Image</summary>
         ![image](https://github.com/YimMenu/YimMenu/assets/139460769/0c1e1feb-78f3-4018-adcb-7ab6c4721484)
       </details>       
- [ ] ability to edit a collection.
- [ ] show green marker on top of selected object/ped. 

Other new features -

- [x] add buttons to do 3 WHEEL LOWRIDER. 
       one can do this using controller. But on pc it is not possible. I have found a way to do it. 
       <details>
         <summary>Image</summary>
         ![image](https://github.com/YimMenu/YimMenu/assets/139460769/7e3a12d0-56d4-46ad-b4cd-96dbd31f5f54)
       </details>
- [x] hide your name from yimmenu.
       <details>
         <summary>Image</summary>
         ![image](https://github.com/YimMenu/YimMenu/assets/139460769/f56dc4e1-cb6e-47cb-a29a-eb62ab178216)
       </details>    
- [x] update on steal outfit functionality 
        - Steal outfit should now able to steal props (such as sunglasses, wrist watch, earpiece etc) as well. Also face and hair will remain unchanged now which is nice.
        - `steal identity` is replaced by `steal outfit`  in context menu for player. 
 - [x] Set random component variations on self. This will set random components only on you.
        <details>
         <summary>Image</summary>
         ![image](https://github.com/YimMenu/YimMenu/assets/139460769/b432710d-d113-4dee-92cc-d40065b484fc)
       </details>

**Comment by lonelybud** [comment]
## Features in more details.

<details>
 <summary>playing scenarios (very simple UI right now)</summary>

  https://github.com/YimMenu/YimMenu/assets/139460769/47c3c746-f6e4-490f-93d1-300f5fda03b5
</details>

<details>
 <summary>spawning, saving, attaching objects</summary>

  https://github.com/YimMenu/YimMenu/assets/139460769/9dcc84b0-a7d4-4b93-9b3f-b2d9895288f5
</details>

<details>
 <summary>creating, spawning collection</summary>

  https://github.com/YimMenu/YimMenu/assets/139460769/b2da2e91-1332-4b7f-b528-089bd99fbf8e
</details>

<details>
  <summary>spawning a track</summary>

  https://github.com/YimMenu/YimMenu/assets/139460769/8a742ad3-9f43-4615-84df-dc96bee3a569
</details>

<details>
 <summary>spawning object/collection on last selected ped/player</summary>
  
  https://github.com/YimMenu/YimMenu/assets/139460769/fb2edd1a-6cf7-40ae-8839-c4a6b89c3cc1
</details>

<details>
 <summary>lifting a car (use left mouse click to throw 1x speed. right muse click to throw 2x speed ). If unable to detach/throw the car then point to that lifted car and click the lift up again in context menu.</summary>

  https://github.com/YimMenu/YimMenu/assets/139460769/df0ddbff-3fcc-4a06-9da0-c0d67b16ee0f
</details>

<details>
 <summary>3 wheel Low rider car </summary>

  https://github.com/YimMenu/YimMenu/assets/139460769/2cf1ee3c-e21b-4dac-be8f-f29fa4b9d650
</details>

<details>
 <summary>Custom tab for particle effects (custom group, name and size can be saved)</summary>

  https://github.com/YimMenu/YimMenu/assets/139460769/e55a9fdf-c1a4-4102-94eb-dc84270ecf38
</details>

>Important notes -
>- remember to delete spawned objects, collection as a good practice (use shift + click). Spawned objects and collections can be deleted by shift clicking them. Collections, objects inside collections can also be shift click deleted.
>- female only play certain scenarios [see](https://www.youtube.com/watch?v=k14GP5QHS1E)

## Files to paste in yim dir

<details>
 <summary>yim dir can be opened as follows (video)</summary>

  https://github.com/YimMenu/YimMenu/assets/139460769/5de0b618-1725-4c22-9fc4-19f2869923d6
</details>

<details>
 <summary>files</summary>

- scenarios list - https://github.com/DurtyFree/gta-v-data-dumps/blob/master/scenariosCompact.json
- objects list - https://github.com/DurtyFree/gta-v-data-dumps/blob/master/ObjectList.ini
- one can create own collection file. But I am providing a sample one [here](https://github.com/YimMenu/YimMenu/files/12323604/savedObjectCollections.zip).
- particles effect list - https://github.com/DurtyFree/gta-v-data-dumps/blob/master/particleEffectsCompact.json
- sample saved particles effect - [savedPtxEffects.zip](https://github.com/YimMenu/YimMenu/files/12431338/savedPtxEffects.zip)
- soundNames list - https://github.com/DurtyFree/gta-v-data-dumps/blob/master/soundNames.json
- sample animations file :) - [ped_animations.zip](https://github.com/YimMenu/YimMenu/files/12550880/ped_animations.zip)
- sample saved audio file - [savedAudios.zip](https://github.com/YimMenu/YimMenu/files/12550886/savedAudios.zip)
</details>



Download and paste file as is it. If its a zip then extract its content.

**Reply by lonelybud** [reply]
https://github.com/dynamoNg/YimMenu/pull/9

build
https://github.com/dynamoNg/YimMenu/actions/runs/5834682542

place this file in yim dir for listing objects -
https://github.com/DurtyFree/gta-v-data-dumps/blob/master/ObjectList.ini

sample collection file:
[savedObjectCollections.zip](https://github.com/YimMenu/YimMenu/files/12323604/savedObjectCollections.zip)

I am currently not submitting it to Yim bcz of poor code quality and not feature complete but please free to play with it and use the code.

**Comment by lonelybud** [comment]
spawned objects and collections can be deleted by shift clicking them.

 collections, objects inside collections can be shift click deleted.

**Reply by alyxme** [reply]
This is great! Realy hope this will be implemented into the main code. The thing to make tranfer animations and scenarios to PEDs is awesome and there would be another feature to add a props them too!

**Reply by lonelybud** [reply]
yes that can be added. But the menu will get cluttered. I hope there can be possibility for another menu or submenu.

**Comment by minecraftdogblowerupper** [comment]
I really like the look of this but I have no idea how to install can you help?

**Reply by lonelybud** [reply]
i will post a video for this.

**Reply by lonelybud** [reply]
posted video in first comment of this discussion

**Reply by lonelybud** [reply]
place this file in yim dir (X:\Users\USERNAME\AppData\Roaming\YimMenu) for listing objects -
https://github.com/DurtyFree/gta-v-data-dumps/blob/master/ObjectList.ini

**Reply by minecraftdogblowerupper** [reply]
okay I figured out where to put it but no new tabs pop up under animations, is there another thing I need to add? (maybe need to recompile???)

**Reply by minecraftdogblowerupper** [reply]
I tried to clone you're version and it didnt work is there a file I need for the spawner tabs???

**Reply by V1RU513N** [reply]
https://github.com/dynamoNg/YimMenu/actions/runs/5834682542

**Reply by minecraftdogblowerupper** [reply]
I aint even gonna lie, I got no Idea what to do with that

**Reply by lonelybud** [reply]
you can download the dll file here https://github.com/dynamoNg/YimMenu/pull/9#issuecomment-1679764061
and use xenos to inject it in gta https://youtu.be/6vj1H_PPhcQ

**Comment by lonelybud** [comment]
two new features added - 

- spawning object/collection on last selected ped/player
- lifting a car (use left mouse click to throw 1x speed. right muse click to throw 2x speed )

video demonstration attached in first comment of this thread.

The new binary zip will be attached when last ci check succeeds.
https://github.com/dynamoNg/YimMenu/pull/9#issuecomment-1679764061

**Comment by EljanSantiago** [comment]
hello, may i know where to get the scenarios json pack? and how can i spawn an object? can u help me to step by step? thankyou so much. im just really so confused

p.s: I already added the listing objects files and sample collection files in directory, but nothing happens thankyou so muchagain for helping me

**Reply by lonelybud** [reply]
please see https://github.com/YimMenu/YimMenu/discussions/1918#discussioncomment-6663110

**Comment by alyxme** [comment]
I'd like to see spawn-prop&anim feature in the menu. Could you please pull this into the Master? In future it would be done much harder with non-related features in the code. Spawn-prop part is works and looks great! Atleast you can put the code in order while in draft state.

**Reply by lonelybud** [reply]
yes you are right but i dont think the menu will accepting `spawn objects & Scenarios ` as in native code. Yes animations was added as native code under animations tab and its fine.

You dont have to worry about multiple features put down in one branch. They still live in separate branches and are merged in one common develop branch. Its true that merge order matters for some branches bcz some shared code was required my another branch.

I wrote these code as fun. Fun stuffs should be added in Lua I guess. I am currently not good at Lua. If i start writing the code in lua i will inform here but cannot promise for now. 

One can use the code to extend it or write it in lua. I know some of the code is native and not present in lua bindings. Its a great effort.

The easiest thing i can do right now is merging any master code from yim in feature branch https://github.com/lonelybud/YimMenu/pull/9

if you have anything you can ask.

**Reply by alyxme** [reply]
I'm sure you can do something useful with [Animation.zip](https://github.com/YimMenu/YimMenu/files/12602086/Animation3.zip) for YimMenu.

**Comment by lonelybud** [comment]
i observed that objects and collections are not deleted. I guess there is an issue with delete functionality. Will check and fix it.

**Reply by lonelybud** [reply]
- fix: spawned objects/collections are not deleted.
- show popup when deleting object in object list, deleting object in collection and deleting collection itself.

**Comment by lonelybud** [comment]
game is crashing when object is attached and we change parameters like rotation when attached.

```
[14:55:12.0667794][WARN/invoker.cpp:56] Exception caught while trying to call 0x6B9BBD38AB0796DF native.
```

crash from other players when spawning near them -
```
[23:04:37.4551098][WARN/notify.hpp:30] Blocked invalid object model crash from XXXX (116473107)
[23:04:37.4551438][WARN/reaction.cpp:29] Received Crash from XXXX (116473107)
```

**Comment by maybegreat48** [comment]
@lonelybud You've locked your PR so I can't comment there, but you should not get in trouble for calling the natives that you mentioned

**Reply by lonelybud** [reply]
thanks. got it.

**Comment by Maxi23-1** [comment]
Hi, does this still work?

**Reply by lonelybud** [reply]
This is pretty old for me. I abandoned this project long ago since the ban. Code still working but i personally dont use it or recommend using. If one is looking to make lua script out if it can check here though -

spawn object
https://github.com/lonelybud/lonelyMenu/blob/fb2d4749f8d044a0c034f2bc45e2afc250f5e3f2/src/views/world/view_spawn_object.cpp

lift and throw car
https://github.com/lonelybud/lonelyMenu/commit/fe4e680b25aecc139b9294d0c0af1946e0b159af

lowrider controls
https://github.com/lonelybud/lonelyMenu/blob/master/src/views/vehicle/view_vehicle.cpp

scenarios
https://github.com/lonelybud/lonelyMenu/blob/fb2d4749f8d044a0c034f2bc45e2afc250f5e3f2/src/views/self/view_scenarios.cpp

if you are looking for spawning maps. yimmenu should have that feature.

**Comment by Luis56655** [comment]
Hello,How can i attach props to player/PED now?

**Reply by Yimura** [reply]
You can’t attach objects to other player peds (anymore).

**Reply by Luis56655** [reply]
<img width="893" alt="屏幕截图 2024-02-14 195238" src="https://github.com/YimMenu/YimMenu/assets/160056078/6ba12a2c-7eaa-4ea0-b35d-81d567457d92">

**Reply by Luis56655** [reply]
Particle effects can be attached to players, why can't objects be attached to players?

**Reply by Luis56655** [reply]
can i attached objects to players like attached Particle effects to players?

**Reply by lonelybud** [reply]
it is not possible to attach objects to player ped. i have tried it before.
